grafana/grafana
# PublicDashboards: Frontend routing for public dashboards

First pass of rendering a public dashboard.

Known Issues:
- If you select a time slice, you can't undo it unless you remove the query params from the url
- You can see the public view of any dashboard. This will be fixed once we have the backend portion done (will return a 404 if the public url doesnt match up with a dashboard or has been revoked).

There are still a few things that are missing since the backend work for them is still in progress:
1. Public dashboard web route that that returns the dashboard definition associated with it. For now I'm loading the dashboard directly.
2. Api endpoint for public panels to get their data.

How to functionally test it:
- Open up a dashboard
- In the browser url, change `/d` to `/p`. Then refresh.
- The dashboard should be loaded without the nav bar and dashboard header. There should be no panel dropdowns and you should not be able to add annotations.
- If you go to this same public url in incognito mode unauthenticated, the only failing request should be the one for the dashboard trying to execute the queries. This will be updated in a later PR.

<img width="1605" alt="Screen Shot 2022-05-19 at 10 57 22 AM" src="https://user-images.githubusercontent.com/20195316/169356228-f818b8b7-bba6-4beb-87a2-ab6887a2a5dc.png">
